core/lib/instrument.c

```diff
@@ -1791,7 +1791,7 @@ create_and_initialize_module_data(app_pc start, app_pc end, app_pc entry_point,
             copy->segments[i].end = os_segments[i].end;
             copy->segments[i].prot = os_segments[i].prot;
         }
-    } else
+    } else if (segments != NULL)
```

Comment:
IMHO the "} else if (segments != NULL)" makes the code hard to understand and makes it look buggy so it needs either a clear comment (maybe also with an else ASSERT_NOT_REACHED()) or the warning fix needs to be done in a different way to keep the code clear.